ui/src/app/components/results/test-case-result-details.component.html

```diff
@@ -69,20 +69,42 @@
             [class.mouse-over]="testCaseResult?.testCase?.preRequisite">
             <button
               [disabled]="startSync"
-              *ngIf="(testCaseResult.iteration && !testCaseResult.parentResult?.isExecuting||!testCaseResult.iteration) && !testCaseResult?.isExecuting && isDry"
+              *ngIf="(testCaseResult.iteration && !testCaseResult.parentResult?.isExecuting||!testCaseResult.iteration) && !testCaseResult?.isExecuting && isDry && activeTab=='steps'"
               (click)=" startExecution(getTestCaseResult()?.testDeviceResult?.testPlanResult?.dryTestPlan)"
               class="theme-btn-primary py-6">
               <i class="fa-filled-circled-play pr-4"></i>
               <span [translate]="startSync ? 'btn.common.please_wait' :'dry.testcase.btn.re_run'"></span>
             </button>
           </div>
+          <button
+            *ngIf="(testCaseResult.iteration && !testCaseResult.parentResult?.isExecuting||!testCaseResult.iteration) && !testCaseResult?.isExecuting && isDry && activeTab =='steps_current'"
```

Comment:
`(testCaseResult.iteration && !testCaseResult.parentResult?.isExecuting||!testCaseResult.iteration)` this condition looks contradictory

Reply:
changed to `(testCaseResult.iteration && !testCaseResult.parentResult?.isExecuting)`

Reply:
changed to `testCaseResult.parentResult?.isExecuting || !testCaseResult.iteration`